index.js

```diff
@@ -77,21 +133,34 @@ genderCount(data);
 // ***************************************************************************
 // Bonus - Iteration 6
 // ***************************************************************************
-
+console.log("===== Iteration 6 ====== ");
 const promo20 = users => {
   // Your code goes here ...
+  users.forEach(user => {
+    let bal = user.balance.replace('$', "").replace(',', "");
+    if (bal > 20000) {
+      console.log(`Dear ${user.firstName},, since your balance is ${user.balance}, you are eligible to apply for this awesome credit card.`);
+    }
+  });
 };
 
+promo20(usersArray);
+
 // expected output:
 // Dear Howard, since your balance is $21,307.75, you are eligible to apply for this awesome credit card.
 // Dear Rachelle, since your balance is $35,121.49, you are eligible to apply for this awesome credit card.
 
 // ***************************************************************************
 // Bonus - Iteration 7
 // ***************************************************************************
-
+console.log("===== Iteration 7 ====== ");
 const addActive = users => {
   // Your code goes here ...
+  [...users].forEach(user => {
+    user.isActive = true;
+    // user.isActive2 = true;
```

Comment:
I love the fact you are making a copy of the users , and therefore trying to to affect them directly. However, you would never get access to this value, because a forEach never returns anything.
You could have mapped and return that value direcrlt

```js
const addActive = users  => {
    return users.map(user => {
        return {
            ...user, //here taking the whole object
            isActive: true
        }
    })
}
```

Hope this makes sense